src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowDiagnosticTests.cs

```diff
@@ -2457,5 +2457,50 @@ public static void Main()
                 Diagnostic(ErrorCode.ERR_UseDefViolation, "k").WithArguments("k").WithLocation(10, 20)
                 );
         }
+
+        [WorkItem(9581, "https://github.com/dotnet/roslyn/issues/9581")]
+        [Fact]
+        public void UsingSelfAssignment()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        using (System.IDisposable x = x)
+        {
+        }
+    }
+}";
+            CSharpCompilation comp = CreateCompilationWithMscorlib(source);
+            comp.VerifyDiagnostics(
```

Comment:
CS1717 is used in Roslyn for the assignment operator and field declarations, but not for local declarations. For local declarations it would be a cascaded diagnostic, as there is always a definite assignment error.

---
In reply to: [79029925](https://github.com/dotnet/roslyn/pull/13807#discussion_r79029925) [](ancestors = 79029925)